### PR TITLE
Create bottom multiline dialog with toggles

### PR DESCRIPTION
**Pull request by Alvareon**
Capture_dialog

### CODE REVIEW

File: Assets/MirageXR/Player/Scripts/Mobile/DialogWindow/DialogViewBottomMultilineToggles.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class DialogViewBottomMultilineToggles : DialogViewBottom
+{
+    [SerializeField] private Button _buttonClose;
+    [SerializeField] private Toggle _togglePrefab;
+    [SerializeField] Color warningColor = Color.red;
+    public override void UpdateView(DialogModel model)
```
Comment (cawke) [inline Assets/MirageXR/Player/Scripts/Mobile/DialogWindow/DialogViewBottomMultilineToggles.cs]:
```suggestion
    [SerializeField] Color warningColor = Color.red;

    public override void UpdateView(DialogModel model)
```

File: Assets/MirageXR/Player/Scripts/Mobile/DialogWindow/DialogViewBottomMultilineToggles.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                }
+
+                if (text.text == DBManager.domain)
+                {
+                    toggle.isOn = true;
+                }
+                else
+                {
+                    toggle.isOn = false;
+                }
```
Comment (cawke) [inline Assets/MirageXR/Player/Scripts/Mobile/DialogWindow/DialogViewBottomMultilineToggles.cs]: will the toggle only work for one case?

**Review by cawke — CHANGES_REQUESTED** [review]
add a bool variable to the `DialogButtonContent` that will mark that the button is checked. use this parameter as the value for the activity of the toggle. 
it only works for one case right now.

File: Assets/MirageXR/Player/Scripts/Mobile/DialogWindow/DialogViewBottomMultilineToggles.cs (hunk 2 above)
Comment (Alvareon) [reply Assets/MirageXR/Player/Scripts/Mobile/DialogWindow/DialogViewBottomMultilineToggles.cs]: fixed

File: Assets/MirageXR/Player/Scripts/Mobile/DialogWindow/DialogViewBottomMultilineToggles.cs (hunk 1 above)
Comment (Alvareon) [reply Assets/MirageXR/Player/Scripts/Mobile/DialogWindow/DialogViewBottomMultilineToggles.cs]: fixed

File: Assets/MirageXR/Tests/NewUI/ProfileView.cs [hunk 3]
```diff
@@ -208,10 +209,15 @@ private static bool IsValidUrl(string urlString)
...
-            ("Other", ShowServerPanel));
+        RootView_v2.Instance.dialog.ShowBottomMultilineToggles("Moodle servers:",
+            (DBManager.WEKIT_URL, () => ChangeServerAndPrivacyPolicyDomain(DBManager.WEKIT_URL, DBManager.WEKIT_PRIVACY_POLICY_URL), false, IsSelectedToggle(DBManager.WEKIT_URL)),
+            (DBManager.ARETE_URL, () => ChangeServerAndPrivacyPolicyDomain(DBManager.ARETE_URL, DBManager.ARETE_PRIVACY_POLICY_URL), false, IsSelectedToggle(DBManager.ARETE_URL)),
+            (CUSTOM_SERVER_TEXT, ShowServerPanel, false, IsSelectedToggle(CUSTOM_SERVER_TEXT)));
+    }
+
+    private bool IsSelectedToggle(string text)
+    {
+        return (text == CUSTOM_SERVER_TEXT && DBManager.domain != DBManager.WEKIT_URL && DBManager.domain != DBManager.ARETE_URL) || (text == DBManager.domain);
```
Comment (cawke) [inline Assets/MirageXR/Tests/NewUI/ProfileView.cs]:
```suggestion

        var isWekitSelected = DBManager.domain == DBManager.WEKIT_URL;
        var isAreteSelected = DBManager.domain == DBManager.ARETE_URL;

        RootView_v2.Instance.dialog.ShowBottomMultilineToggles("Moodle servers:",
            (DBManager.WEKIT_URL, () => ChangeServerAndPrivacyPolicyDomain(DBManager.WEKIT_URL, DBManager.WEKIT_PRIVACY_POLICY_URL), false, isWekitSelected),
            (DBManager.ARETE_URL, () => ChangeServerAndPrivacyPolicyDomain(DBManager.ARETE_URL, DBManager.ARETE_PRIVACY_POLICY_URL), false, isAreteSelected),
            (CUSTOM_SERVER_TEXT, ShowServerPanel, false, !(isWekitSelected || isAreteSelected));
```

**Review by cawke — CHANGES_REQUESTED** [review]

**Comment by cawke** [comment]
I'll check after the merge #1346

File: Assets/MirageXR/Tests/NewUI/ProfileView.cs [hunk 4]
```diff
@@ -278,9 +282,9 @@ private void OnClickPrivacyPolicy()
 
     private void UpdatePrivacyPolicyButtonActive()
     {
-        var setActive = (DBManager.privacyPolicyDomain != string.Empty) ? true : false;
+        //var setActive = (DBManager.privacyPolicyDomain != string.Empty) ? true : false;
 
-        _btnPrivacyPolicy.gameObject.SetActive(setActive);
+        //_btnPrivacyPolicy.gameObject.SetActive(setActive);
```
Comment (cawke) [inline Assets/MirageXR/Tests/NewUI/ProfileView.cs]: these changes should be here?

**Review by cawke — CHANGES_REQUESTED** [review]

File: Assets/MirageXR/Tests/NewUI/ProfileView.cs (hunk 4 above)
Comment (Alvareon) [reply Assets/MirageXR/Tests/NewUI/ProfileView.cs]: not necessarily, it is discussed in the [PR_1347](https://github.com/WEKIT-ECS/MIRAGE-XR/pull/1346)

**Review by cawke — APPROVED** [review]
ty

**Pull request merged**